### DISCUSSION TITLE
The future direction of development

### DISCUSSION THREAD

**Discussion by itoldya** [benhowell/react-grid-gallery#179] [General]
Hi, I'm Igor 👋 

I’ve recently been working on resolving existing issues and have to admit that most of them cannot be resolved without serious changes.

The number one problem now is the external dependency on `react-images`.
The current version we use isn't working correctly on React >16. There is a new version of the package which I suppose works on React 17, but [it's marked as not recommended to use](https://github.com/jossmac/react-images) in the new projects.

I guess it's possible to find a stable replacement for `react-images`, but of course, we cannot change the lightbox provider seamlessly. It will cause our API to change because we are dependent on exact implementation.
Also, we have to take risks that the replacement can also become out of date one day.
And even with a new lightbox, it will be not easy to provide customization asked in the issues.

So that's why I want to suggest dropping lightbox functionality out of `react-grid-gallery` and instead focusing on building a beautiful justified grid library with selection functionality. 

It'll give us the way to upgrade to the latest React version and resolve all warnings. Also, it means
- more flexibility, users can use any lightbox they want or don't use if they don't need it
- component API simplification, getting rid of proxy props
- tiny package size
- we are aligned with JS ecosystem philosophy and can be focused on the thing we are good at.

As the library focused on the justified grid we would implement new features in the future such as 
- new grid arrangement options (for example original google photos grid is also justified but it has no fixed row heights and can scale images)
- adding animation on selection
- export utils we use for building grids for people to render their custom galleries or even use it without React.

Of course, I understand that some users still wanted to use a grid with a lightbox, so I suggest adding a docs section with examples of integrations for different use cases
- with popular lightbox libraries
- infinity scroll gallery.

So what do you think of this?

I understand this repo had been abandoned for a while and there is no active community around React Grid Gallery. But don't let it stop you from leaving your opinion here. I really appreciate it. Thanks!

**Comment by benhowell** [comment]
Hi Igor,
Thanks for starting the discussion!

I agree with everything you've suggested. Dropping the lightbox functionality would be the best way forward in my opinion and would make for a small, efficient and clean library with a concise API, like you say. 

Adding lightbox functionality as an external call (react callback/hook/whatever) would be trivial for the user to implement. 

Once react-images is stripped out the library, we can probably bump react-grid-gallery versioning up to 1.x.x.

**Comment by haschdl** [comment]
Thank you @itoldya  and @benhowell for putting some much TLC in the project! I also add +1 to drop functionality in favor of ensuring the maintainability and future for the component.  However this seems to be already "done"?

**Reply by itoldya** [reply]
Thanks! Yes, it's done.
